test/e2e/certificate/certificate_ca.go

```diff
@@ -64,11 +64,45 @@ var _ = framework.CertManagerDescribe("CA Certificate", func() {
 		secretClient := f.KubeClientSet.CoreV1().Secrets(f.Namespace.Name)
 
 		By("Creating a Certificate")
-		_, err := certClient.Create(util.NewCertManagerBasicCertificate(certificateName, certificateSecretName, issuerName, v1alpha1.IssuerKind))
+		_, err := certClient.Create(util.NewCertManagerBasicCertificate(certificateName, certificateSecretName, issuerName, v1alpha1.IssuerKind, 0, 0))
 		Expect(err).NotTo(HaveOccurred())
 		By("Verifying the Certificate is valid")
 		err = util.WaitCertificateIssuedValid(certClient, secretClient, certificateName, time.Second*30)
 		Expect(err).NotTo(HaveOccurred())
 	})
 
+	cases := []struct {
+		inputDuration    time.Duration
+		inputRenewBefore time.Duration
+		expectedDuration time.Duration
+		label            string
+	}{
+		{
+			inputDuration:    time.Hour * 24 * 35,
+			inputRenewBefore: 0,
+			expectedDuration: time.Hour * 24 * 35,
+			label:            "35 days",
+		},
+		{
+			inputDuration:    0,
+			inputRenewBefore: 0,
+			expectedDuration: time.Hour * 24 * 90,
+			label:            "the default duration (90 days)",
+		},
+	}
+	for _, v := range cases {
+		v := v
+		It("should generate a signed keypair valid for "+v.label, func() {
+			certClient := f.CertManagerClientSet.CertmanagerV1alpha1().Certificates(f.Namespace.Name)
+			secretClient := f.KubeClientSet.CoreV1().Secrets(f.Namespace.Name)
+
+			By("Creating a Certificate")
+			_, err := certClient.Create(util.NewCertManagerBasicCertificate(certificateName, certificateSecretName, issuerName, v1alpha1.IssuerKind, v.inputDuration, v.inputRenewBefore))
+			Expect(err).NotTo(HaveOccurred())
+			By("Verifying the Certificate is valid")
+			err = util.WaitCertificateIssuedValid(certClient, secretClient, certificateName, time.Second*30)
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
I think we're missing a call to `f.CertificateDurationValid(cert, v.expectedDuration)`?